### ISSUE TITLE
Cannot access the CF API endpoint

### ISSUE THREAD

**Issue by StanleyShen** [cloudfoundry/cf-release#847]
Hello, all

I am trying to deploy CF to AWS using the bosh-lite.
Normally the api endpoint is api.bosh-lite.com which points to a local IP address like 10.244

If I need to access it via xip.io domain name, I need to create a stub file to override the domain to PUBLIC_IP_OF_LITE_VM.xip.io and generate the cf.yml

from generated yml I can see the domain name are changed to the xip.io related.
But after deploying the CF, I still cannot access the API endpoint, can anyone help on it?

when launching the vagrant VM, the related parameters are:

export BOSH_AWS_ACCESS_KEY_ID=<key id>
export BOSH_AWS_SECRET_ACCESS_KEY=<key value>
export BOSH_LITE_SECURITY_GROUP=sg-d63f72b2
export BOSH_LITE_PRIVATE_KEY=/home/ubuntu/.ssh/id_rsa
export BOSH_LITE_KEYPAIR=bosh
export BOSH_LITE_NAME=vagrant

I also assigned a elastic ip to the bosh lite VM.

If I try to "telnet liteVMIP 443", it says " Connection refused".
The 443 port is opened already in the security group.

**Comment by Amit-PivotalLabs** [comment]
Can you confirm that you have:

- the BOSH-Lite VM you created with vagrant has an elastic IP actually associated to it
- it belongs to a security group that allows incoming TCP connections to port 443 from any source IP
- `telnet BOSH_LITE_ELASTIC_IP 443` fails

Next, are you deploying to a VPC? Or plain EC2?  If you're deploying to a VPC, then you need to make sure that your BOSH-Lite VM is in a publicly routable subnet of your VPC, namely one with an Internet Gateway as the default route in its associated route table.

**Comment by StanleyShen** [comment]
Thanks for reply.

Yes it's deploying to VPC,and I create the VPC based on http://bosh.io/docs/init-aws.html
And here are my network related settings, and I think it's already there regarding your question?
Is there anything wrong here?

VPC ID:   vpc-7cedb719 | bosh
Network ACL: acl-ed664388
State: available
Tenancy: Default
VPC CIDR: 10.0.0.0/16
DNS resolution: yes
DHCP options set: dopt-9bc6dff9
DNS hostnames: yes
Route table: rtb-1c2e0079



Subnet ID: subnet-ddf5bbaa | public
Availability Zone: us-west-2a
CIDR: 10.0.0.0/24
Route table: rtb-1f2e007a
State:available
Network ACL:acl-ed664388
VPC: vpc-7cedb719 (10.0.0.0/16) | bosh
Default subnet: no
Available IPs: 249
Auto-assign Public IP: yes

Route Table ID: rtb-1f2e007a
Main: no
Explicitly Associated With: 1 Subnet
VPC: vpc-7cedb719 (10.0.0.0/16) | bosh

Destination   Target    Status  Propagated
10.0.0.0/16   local     Active  No
0.0.0.0/0  igw-432fb726	Active  No

ID: igw-432fb726
Attached VPC ID: vpc-7cedb719 (10.0.0.0/16) | bosh
State: attached
Attachment state: available

**Comment by StanleyShen** [comment]
Attach instance info and security group too.

Instance:
Instance ID : i-0bab71d2
Public DNS : ec2-52-25-158-120.us-west-2.compute.amazonaws.com
Instance state : running
Public IP : 52.25.158.120
Instance type : m3.xlarge
Elastic IP  : 52.25.158.120
Private DNS :ip-10-0-0-245.us-west-2.compute.internal
Availability zone:  us-west-2a
Private IPs :10.0.0.245
Security groups : bosh-lite. view rules
Secondary private IPs
Scheduled events:  No scheduled events
VPC ID : vpc-7cedb719
AMI ID : boshlite-9000.69.0 (ami-de5f49bf)
Subnet ID : subnet-ddf5bbaa
Platform -

Security Group:
HTTP     			TCP 	 80             0.0.0.0/0   
All TCP  			TCP 	0 - 65535       sg-d63f72b2 (bosh-lite)
All traffic			All 	All  			sg-d63f72b2 (bosh-lite)
SSH 				TCP 	22				0.0.0.0/0
Custom TCP Rule 	TCP		25555			0.0.0.0/0
All UDP 			UDP		0 - 65535		sg-d63f72b2 (bosh-lite)
Custom TCP Rule 	TCP		4443			0.0.0.0/0
HTTPS				TCP		443				0.0.0.0/0

**Comment by Amit-PivotalLabs** [comment]
And just to confirm, those are the *inbound* traffic rules for the security group, right?

**Comment by StanleyShen** [comment]
yes, they are at the inbound traffic rules.

**Comment by Amit-PivotalLabs** [comment]
This excerpt from [here](https://github.com/cloudfoundry/bosh-lite/blob/master/docs/aws-provider.md) may be relevant:

> As part of Vagrant provisioning bosh-lite is setting IP tables rules to direct future traffic received on the instance to another IP (the HAProxy). These rules are cleared on restart. In case of restart they can be created by running vagrant provision.

I can telnet to 25555 and 22, just not 443 and 80.  443 and 80 connections are meant to be handled by the HA Proxy process running within a container inside the BOSH-Lite VM, so the fact that this fails to telnet suggests either that process is down or the iptables rule forwarding traffic on those ports to the container need to be re-established.  I am able to see that the HA Proxy process is running fine, so perhaps this is the issue.

BTW, you may wish to change your director credentials.  See https://bosh.io/docs/director-users.html

**Comment by StanleyShen** [comment]
Thanks, I just had a try, after running the vagrant provision, I can telnet the 443 port.
As I changed the domain name to bosh-lite.com yesterday, I will have another try for the xip.io later. (should work as 443 can be telnet)
For director credentials, I will change them if I get the deployment well.
Thank a lot for asking the questions, will update it again later.

**Comment by StanleyShen** [comment]
Tried it, it works.
And the problem should be that:
when I ran vagrant up --provider=aws  the vagrant is hang there and it didn't ssh to the bosh-lite VM to set the iptables rule forwarding.
I generated the id_ras public/private keys, and I replaced the bosh private key to id_rsa file.
But I cannot ssh to bosh-lite VM using ssh ubuntu@bosh-lite
If I change the id_ras to new name like id_ras.bosh and update the environment for it and then I can ssh to bosh-lite VM.

I am new on it, thanks for help, I appreciate that.
